core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -260,11 +262,13 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
         this.shareGroupMetrics.shareAcknowledgement();
         Map<TopicIdPartition, CompletableFuture<Errors>> futures = new HashMap<>();
         acknowledgeTopics.forEach((topicIdPartition, acknowledgePartitionBatches) -> {
-            SharePartition sharePartition = partitionCacheMap.get(sharePartitionKey(groupId, topicIdPartition));
+            SharePartitionKey sharePartitionKey = sharePartitionKey(groupId, topicIdPartition);
+            SharePartition sharePartition = partitionCacheMap.get(sharePartitionKey);
             if (sharePartition != null) {
                 CompletableFuture<Errors> future = new CompletableFuture<>();
                 sharePartition.acknowledge(memberId, acknowledgePartitionBatches).whenComplete((result, throwable) -> {
                     if (throwable != null) {
+                        handleSharePartitionException(sharePartitionKey, throwable);
```

Comment:
To be consistent, we want to add the same logic for shareFetch too.

To do this, we need to extend FetchParams such that `fetchOnlyLeader()` is true for share fetch and handle `NotLeaderOrFollowerException` accordingly.

Reply:
Done.